src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusExporter.cs

```diff
@@ -32,22 +33,24 @@ public PrometheusExporter(PrometheusExporterOptions options)
         this.CollectionManager = new PrometheusCollectionManager(this);
     }
 
+    public delegate ExportResult ExportFunc(in Batch<Metric> batch);
```

Comment:
I saw that this was referenced in other classes. was the point of defining a custom delegate to contractually require the `in` keyword? :)

Reply:
> was the point of defining a custom delegate to contractually require the in keyword? :)

💯 ya. More precisely, we want `Batch<T>` to be passed by reference. Because it is a large[ish] value type/struct. This is a perf optimization although I'll admit, a very minor one. The time spent passing `Batch<T>` by value (copy) is likely dwarfed by the time spent writing out the metrics.